### PR TITLE
enhancement

### PR DESCRIPTION
**Pull request by arturocuma**
* ASimulationState: Add TMap<FString, std::string> EncodedStrings to handle Japanese txt
* URRBaseLidarComponent::LidarID
* URRMathUtils::ClampAngle() use FMod directly without NormalizeAxis()/ClampAxis() to avoid unintentional sign changing

### CODE REVIEW

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp [hunk 1]
```diff
@@ -85,6 +85,19 @@ void URRFloatingMovementComponent::TickComponent(float InDeltaTime,
                                      bSweepEnabled,
                                      hit);
 
+#if RAPYUTA_FLOAT_MOVEMENT_DEBUG
+            if (deltaRot.Yaw > 0.f)
+            {
+                UE_LOG_WITH_INFO(LogRapyutaCore,
+                                 Warning,
```
Comment (yuokamoto) [inline Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp:92]: Verbose

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp [hunk 2]
```diff
@@ -85,6 +85,19 @@ void URRFloatingMovementComponent::TickComponent(float InDeltaTime,
...
+                                 TEXT("deltaRot.Yaw: %f, AngularVelocity.Yaw: %f, MaxAngularSpeed: %f, inDeltaTime: %f"),
+                                 deltaRot.Yaw,
+                                 AngularVelocity.Yaw,
+                                 MaxAngularSpeed,
+                                 InDeltaTime);
+            }
+#endif
+
 #if RAPYUTA_FLOAT_MOVEMENT_DEBUG
```
Comment (yuokamoto) [inline Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp]: please merge two #if block?

File: Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h [hunk 3]
```diff
@@ -298,10 +297,9 @@ class RAPYUTASIMULATIONPLUGINS_API URRRobotROS2Interface : public UObject
     void MakeServiceRequest(const FName& InServiceName, const TServiceRequest& InRequest)
     {
         // Create and update request
-        auto* clienPtr = ServiceClientList.Find(InServiceName);
-        if (clienPtr)
+        if (auto* clientPtr = ServiceClientList.Find(InServiceName))
         {
-            UROS2ServiceClient* client = *clienPtr;
+            UROS2ServiceClient* client = *clientPtr;
```
Comment (yuokamoto) [inline Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h]: Do you need to define new var here?

File: Source/RapyutaSimulationPlugins/Public/Sensors/RRBaseLidarComponent.h [hunk 4]
```diff
@@ -67,6 +64,9 @@ class RAPYUTASIMULATIONPLUGINS_API URRBaseLidarComponent : public URRROS2BaseSen
     UFUNCTION(BlueprintCallable)
     void GetData(TArray<FHitResult>& OutHits, float& OutTime) const;
 
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
```
Comment (yuokamoto) [inline Source/RapyutaSimulationPlugins/Public/Sensors/RRBaseLidarComponent.h]: this is not used

File: Source/RapyutaSimulationPlugins/Public/Core/RRMathUtils.h [hunk 5]
```diff
@@ -121,8 +121,10 @@ class RAPYUTASIMULATIONPLUGINS_API URRMathUtils : public UBlueprintFunctionLibra
     template<typename T>
     FORCEINLINE static T ClampAngle(T InAngle, const T InMaxAngle)
     {
-        const T normAngle = FRotator::NormalizeAxis(InAngle);
-        return FMath::Clamp(normAngle, -InMaxAngle, InMaxAngle);
+        // returns Angle in the range (-360,360)
```
Comment (yuokamoto) [inline Source/RapyutaSimulationPlugins/Public/Core/RRMathUtils.h:124]: please update doxygen comment as well.

File: Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h (hunk 3 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h]: This change only renamed an existing variable that had a typo `clienPtr` to `clientPtr`

Comment (yuokamoto) [reply Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h]: My question is do you need to define `client`? can't you directly use clientPtr?

File: Source/RapyutaSimulationPlugins/Public/Core/RRMathUtils.h (hunk 5 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Public/Core/RRMathUtils.h:124]: OK

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp (hunk 1 above)
Comment (Tadinu) [reply Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp:92]: This is already disabled in #if Debug?

File: Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h (hunk 3 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h]: I think we need it, @Tadinu could you help to explain this?

Comment (Tadinu) [reply Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h]: So [TMap::Find()](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Containers/Map.h#L525) returns `ValueType*`, which in this case translates to `UROS2ServiceClient**` thus requires another dereferencing, which However could be also obviated by using `FindRef()` instead like:
```
if (auto* client = ServiceClientList.FindRef(InServiceName))
{
}
```

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp (hunk 1 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp:92]: Yes it its already in` #if RAPYUTA_FLOAT_MOVEMENT_DEBUG`

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp (hunk 2 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp]: OK

File: Source/RapyutaSimulationPlugins/Public/Sensors/RRBaseLidarComponent.h (hunk 4 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Public/Sensors/RRBaseLidarComponent.h]: I will remove it

File: Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h (hunk 3 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h]: Then I will change it to 
```
if (auto* client = ServiceClientList.FindRef(InServiceName))
{
}
```

**Review by yuokamoto — APPROVED** [review]

**Pull request merged**